### PR TITLE
[FE][ADD] add third party login

### PR DESCRIPTION
**Pull request by MarcPIster**
### 🛠 Changes being made

Add third party login

### ✨ What's the context?

Added facebook and google native login
you need to build the apk natively

### 🧠 Rationale behind the change

we want to login via third party services 

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: visitorMobile/src/components/Login/GoogleLogin/GoogleLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import {TouchableOpacity, View, Image, Alert} from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useTranslation } from 'react-i18next';
+import { getVisitorProfileDetails } from '../../../services/profileCalls';
+import {GoogleSignin} from "@react-native-google-signin/google-signin";
+import {loginUser} from "../../../services/userCalls";
+import styles from "./styles";
+
+const GOOGLE_CLIENT_ID = '251768771069-7v10qin71ssbe5au77dt6at41s6cgglg.apps.googleusercontent.com';
+const WEB_CLIENT_ID = '251768771069-9iate5ptc3pm51k380d2uamf3segvtf6.apps.googleusercontent.com';
```
[the remark this answers is not in the document]
Comment (MarcPIster) [reply visitorMobile/src/components/Login/GoogleLogin/GoogleLogin.tsx:11]: No its a public key (We could move it, but why)

File: visitorMobile/App.tsx [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 import { View, Text, StyleSheet, Platform, NativeModules } from 'react-native';
 import NetInfo from "@react-native-community/netinfo";
 import Router from './Router';
 import './i18n/i18n';
 import { useTranslation } from "react-i18next";
+import { GestureHandlerRootView } from 'react-native-gesture-handler';
+import { Settings, LoginManager } from 'react-native-fbsdk-next';
+
+//Settings.setAppID('3681976368725984');
+//Settings.initializeSDK();
```
[the remark this answers is not in the document]
Comment (MarcPIster) [reply visitorMobile/App.tsx:12]: We need it if we build the apk, It will crash without.
Other way, if you try to use expo go with it, it will also crash.

File: visitorMobile/Router.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ import AboutUsScreen from './src/pages/AboutUs/AboutUs';
 import ContactUsScreen from './src/pages/ContactUs/ContactUs';
 import MapPage from './src/pages/MapPage/MapPage';
 import Register from "./src/pages/ProfileScreen/Register/Register";
+//import LoginScreen from "./src/pages/ProfileScreen/Login/LoginWithThridParty";
```
[the remark this answers is not in the document]
Comment (MarcPIster) [reply visitorMobile/Router.tsx:18]: Will add comments why its better to have this there :)

**Review by JosefineM2000 — APPROVED** [review]
